File: LifeSamadhan/Backend/Models/ProviderLocation.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace LifeSamadhan.API.Models
{
    [Table("providerlocations")]
    public class ProviderLocation
    {
        [Key]
        [Column("Id")]
        public long Id { get; set; }

        [Required(ErrorMessage = "ProviderId is required")]
        [Column("ProviderId")]
        public long ProviderId { get; set; }

        [Required(ErrorMessage = "LocationId is required")]
        [Column("LocationId")]
        public long LocationId { get; set; }

        [Required(ErrorMessage = "Status is required")]
        [RegularExpression("^(PENDING|ACTIVE|REJECTED)$",
            ErrorMessage = "Status must be PENDING, ACTIVE or REJECTED")]
        [Column("Status")]
        public string Status { get; set; } = "PENDING";

        [Required(ErrorMessage = "EffectiveFrom date is required")]
        [Column("EffectiveFrom")]
        public DateTime EffectiveFrom { get; set; } = DateTime.UtcNow;

        [Column("EffectiveTo")]
        public DateTime? EffectiveTo { get; set; }

        [ForeignKey(nameof(ProviderId))]
        public ServiceProvider? Provider { get; set; }

        [ForeignKey(nameof(LocationId))]
        public Location? Location { get; set; }
    }
}

File: lifeFull/SpringBootLifeSamadhan/src/main/java/com/lifesamadhan/api/model/ProviderLocation.java
package com.lifesamadhan.api.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import lombok.Builder;

@Entity
@Table(name = "provider_locations", uniqueConstraints = @UniqueConstraint(columnNames = { "provider_id",
        "location_id" }))
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class ProviderLocation {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @NotNull(message = "ProviderId is required")
    @Min(value = 1, message = "ProviderId must be a valid value")
    @Column(name = "provider_id", nullable = false)
    private Long providerId;

    @NotNull(message = "LocationId is required")
    @Min(value = 1, message = "LocationId must be a valid value")
    @Column(name = "location_id", nullable = false)
    private Long locationId;

    @Pattern(regexp = "^(ACTIVE|INACTIVE|PENDING|APPROVED)$", message = "Status must be ACTIVE, INACTIVE, PENDING, or APPROVED")
    @Builder.Default
    @Column(nullable = false)
    private String status = "ACTIVE";

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "provider_id", insertable = false, updatable = false)
    @com.fasterxml.jackson.annotation.JsonIgnore
    private ServiceProvider provider;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "location_id", insertable = false, updatable = false)
    private Location location;
}
